content/en-us/build/stateless/language-basic.md

```diff
@@ -0,0 +1,74 @@
+---
+title: "Language Basics"
+weight: 1002
+menu:
+  build:
+    parent: "phat-stateless"
+---
+
+The Phat Contract is written in Ink! language, which is extended from Rust language (more language supports like Python and Nodejs will be provided in the future). It is finally compiled to WebAssembly (WASM), which also takes security into consideration when designed, then you can upload and deploy them.
+
+Rust is thought to be a hard language, but no worry! It deserves the efforts since Rust is fast and helps prevent many million-dollar bugs. Also I sincerely believe one day you will find it charming since all your enemy is just the Rust compiler itself😼.
+
+The prospering ecosystem is another reason to use Rust-based Ink! to develop your contract. There are many secure and high-performance libraries for you to use without building all the wheels from scratch.
+
+This section will not teach your about the Rust and Ink! language itself. Instead, it provides some helpful links and comments to get you prepared quickly.
+
+
+## Rust
+
+### Tooling
+
+When using Rust, you will always rely on the following official tools to manage your toolchain and compile your projects:
+- [rustup](https://rustup.rs/), which helps you install and manage different versions of Rust compilers;
+- cargo, the package manager installed automatically along with Rust compiler. Each Rust project contains a `Cargo.toml` file to describe its metadata and dependencies, and `cargo` downloads the dependencies and builds the project for you. The [cargo book](https://doc.rust-lang.org/cargo/reference/index.html) can help, and ensure you understand the [[dependencies]](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html) and [[features]](https://doc.rust-lang.org/cargo/reference/features.html) sections in `Cargo.toml` before going on.
+
+### Language Basics
+
+The best material is surely the official [Rust book](https://doc.rust-lang.org/book/).
+- For a beginner, to learn the first 6 sections is enough for your journey in the Phat Contract world;
+- I also recommend to finish the 7 - 10 sections when you decide to go a little deeper.
+
+There are also many [books recommended by the community](https://www.reddit.com/r/rust/comments/sjclfb/best_book_to_learn_rust/) for more advanced topics.
+
+
+## Ink!
+
+Ink! is the programming language developed by Parity. The native Ink! contracts can run on different blockchains (like Polkadot and Phala), but Phala adds its secret ingredient called [pink-extension](https://github.com/Phala-Network/phala-blockchain/tree/master/crates/pink) (short for Phala Ink! Extension) to form Phat Contract which is only functional in Phala's off-chain Workers.
+
+> Phat Contract is the superset of Ink! contract.
+
+### Ink! Macros
```

Comment:
Besides the macros, it's also important to talk about the **concepts** like account, contract address, cross-contract call, etc

Reply:
We shall talk about these in chain-related opertions section.